libs/types/src/cdf/election-results-reporting/convert.ts

```diff
@@ -0,0 +1,269 @@
+import { assertDefined, err, ok, Result } from '@votingworks/basics';
+import getDeepValue from 'lodash.get';
+import { Candidate, CandidateId } from '../../election';
+import { LanguageCode } from '../../language_code';
+import * as ResultsReporting from '.';
+import * as VxTabulation from '../../tabulation';
+import {
+  BallotMeasureContest,
+  CandidateContest,
+  CountItemType,
+  PartyContest,
+  RetentionContest,
+} from '.';
+import {
+  isBallotMeasureContest,
+  isCandidateContest,
+  isRetentionContest,
+} from './types';
+
+type CandidateNameRecord = Record<CandidateId, Candidate['name']>;
+
+function assertGetDeepValue(root: unknown, valuePath: string) {
+  return assertDefined(
+    getDeepValue(root, valuePath),
+    `Couldn't get value: ${valuePath}`
+  );
+}
```

Comment:
I'm a bit worried that using this `getDeepValue` function bypasses type checking, which may make it easier to make a mistake if we ever change this code. Do you feel like there's a tradeoff that makes it worthwhile?

Reply:
I also didn't like the bypassed type checking. Even losing tab completion when adding this feature it was a little annoying.

If we don't use `getDeepValue` then we would use optional chaining `?`. We'd need to handle `undefined` where realistically we'd never see that specific prop be `undefined` eg. `Candidate` on a `CandidateSelection` object. We'd also need to add coverage for those unrealistic `undefined` cases or ignore lines with optional chaining from coverage.

If we wanted to keep type safety and coverage, we could break out the existence checks to avoid `?` altogether. We'd get those 2 benefits at the cost of lots of boilerplate in the code.

IMO losing type safety in this context isn't _as_ bad as normal because we only use `getDeepValue` on objects in the ERR schema, and that schema is unlikely to change. But agreed it's still not ideal.

I added `assertGetDeepValue` pretty early in development so I'm not sure how messy the code will get if we switch to optional chaining, but I can experiment with it to see which feels better.

Reply:
> We'd need to handle undefined where realistically we'd never see that specific prop be undefined
In this case, maybe `assertDefined` instead of optional chaining?

Reply:
https://github.com/votingworks/vxsuite/pull/5344/commits/65424132fcfc26c83553d11402f721cb46e9e4d4